### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by 29td**
### Project milestone: set up associations

_**In this PR the following was implemented:**_

- Created a class Classroom
- Created the has-many/belongs-to relationship between Classroom and Student
- Created a class Book
- Created a class Rental
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: Student.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 class Student < Person
   attr_accessor :classroom
 
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+  def initialize(age, classroom = 5, name = 'Unknown', parent_permission: true)
     super(name, age, parent_permission)
     @classroom = classroom
   end
```
Comment (1GORDON) [inline Student.rb:9]: Great job @29td 🤝

- [x] Kindly add a `belongs-to` method `(a student belongs to a classroom)`. Check this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for more guidance.

File: README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Open terminal and run this command ruby class.rb
 
 ## Author
 
-👤 **Nonhlanhla Mndebele**
+👤 **Nonhlanhla R Mndebele**
 
 - GitHub: [@Nonhlanhla](https://https://github.com/29td)
```
Comment (1GORDON) [inline README.md]: Great job @29td 🤝

- [x] Kindly fix your GitHub handle. The link seems to be broken.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @29td ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Person.rb [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
   end
 
   def can_use_services?
     of_age? >= 18 || @parent_permission
   end
 
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (BenMukebo) [inline Person.rb]: - [OPTIONAL] Great job 👏🏻 But the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association eg:

```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)
 end
```

**Review by BenMukebo — APPROVED** [review]
### Hi @29td , 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Descriptive README 🥇 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by 29td** [comment]
hy @BenMukebo 
Thank you 👍

**Pull request merged**